### PR TITLE
build: Update husky to v9

### PR DESCRIPTION
**Pull request by vinayakkulkarni**
**Please describe the changes this PR makes and why it should be merged:**
- [x] Followed the [migration guide](https://github.com/typicode/husky/releases/tag/v9.0.1) provided by Husky to improve DX 

**Status and versioning classification:**
- This PR includes non-code changes, mostly affects improved DX.

### CODE REVIEW

File: .husky/commit-msg [hunk 1]
```diff
@@ -1,4 +1,3 @@
 #!/bin/sh
```
Comment (vladfrangu) [inline .husky/commit-msg:1]: Should probably remove this line too (I see you removed it in the other hook)

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -30,7 +30,8 @@
 		"Amish Shah <amishshah.2k@gmail.com>",
 		"Vlad Frangu <kingdgrizzle@gmail.com>",
 		"SpaceEEC <spaceeec@yahoo.com>",
-		"Aura Román <kyradiscord@gmail.com>"
+		"Aura Román <kyradiscord@gmail.com>",
+		"Vinayak Kulkarni <inbox.vinayak@gmail.com>"
```
Comment (almostSouji) [inline package.json:34]: this section is for the core maintainers of the project.

```suggestion
		"Aura Román <kyradiscord@gmail.com>"
```

**Review by almostSouji — CHANGES_REQUESTED** [review]

File: .husky/commit-msg (hunk 1 above)
Comment (vinayakkulkarni) [reply .husky/commit-msg:1]: Kept it cause we have `$1` (shell syntax) and having the hashbang allows the IDE to understand that it's shell script

The husky shell script execution is removed as suggested by the migration guide

File: package.json (hunk 2 above)
Comment (vinayakkulkarni) [reply package.json:34]: Cool, will remove it 😃

File: .husky/commit-msg (hunk 1 above)
Comment (vladfrangu) [reply .husky/commit-msg:1]: Then it should be brought back in the other shell script

File: package.json (hunk 2 above)
Comment (vinayakkulkarni) [reply package.json]: - [x] Done in 199b242

File: .husky/commit-msg (hunk 1 above)
Comment (vinayakkulkarni) [reply .husky/commit-msg]: - [x] nevermind, removed the hashbang in 9bf20c6

**Review by sdanialraza — COMMENTED** [review]
Looking at the [husky release v9.0.1](https://github.com/typicode/husky/releases/tag/v9.0.1), can https://github.com/vinayakkulkarni/discord.js/blob/9bf20c6a299f2d804290d9b2543038464a20df8f/package.json#L22 be changed from `is-ci || husky install` to `is-ci || husky`?

**Review by almostSouji — APPROVED** [review]

**Review by didinele — APPROVED** [review]

**Comment by vinayakkulkarni** [comment]
> Looking at the [husky release v9.0.1](https://github.com/typicode/husky/releases/tag/v9.0.1), can [vinayakkulkarni/discord.js@`9bf20c6`/package.json#L22](https://github.com/vinayakkulkarni/discord.js/blob/9bf20c6a299f2d804290d9b2543038464a20df8f/package.json#L22) be changed from `is-ci || husky install` to `is-ci || husky`?

- Done in [ba801f2](https://github.com/discordjs/discord.js/pull/10102/commits/ba801f2b800098c01692470ccf0ac8f173f53ee9)

**Review by iCrawl — APPROVED** [review]

**Review by almeidx — APPROVED** [review]

**Pull request merged**